src/templates/default.js

```diff
@@ -1,5 +1,16 @@
 const {wrapper, renderMarkdown, html} = require("./shared");
 
-module.exports = (page, metaIndex) => wrapper(page, metaIndex, html`
-  ${renderMarkdown(page._md, metaIndex)}
-`);
+module.exports = (page, metaIndex) => {
+  const htmlDoc = wrapper(page, metaIndex, html`
+    ${renderMarkdown(page._md, metaIndex)}
+  `);
+
+  const searchDoc = {
+    path: page._path,
+    title: page.title,
+    text: renderMarkdown(page._md, metaIndex, true),
+    keywords: (page.keywords || []).join(" ")
+  };
```

Comment:
Each page template now has two jobs:

* Render the HTML (like before)
* "Render" the searchable fields

This allows page templates to have great control over searchable terms, like if they are adding sections or have special keywords to provide (e.g. tag fields). I haven't taken advantage of this feature yet but the groundwork's laid.